app/routes/application.js

```diff
@@ -30,29 +30,47 @@ export default TravisRoute.extend(BuildFaviconMixin, KeyboardShortcuts, {
   },
 
   activate() {
-    let repos;
-    if (!this.get('features.proVersion')) {
-      repos = this.get('store').peekAll('repo');
+    this.setupRepoSubscriptions();
+  },
+
+  // We send pusher updates through user channels now and this means that if a
+  // user is not a collaborator of a repo or a user is not signed in, we need to
+  // use repo channels for updates for each repo. This method ensures that a
+  // visitor is subscribed to all of the public repos in the store as long as
+  // they're not a collaborator. It also sets up an observer to subscribe to any
+  // new repo that enters the store.
+  setupRepoSubscriptions() {
+    this.get('store').filter('repo', null,
```

Comment:
So this doesn't make a request itself, because `null` is passed, but the store is still being observed as other components/routes fetch this data, correct? Just want to make sure I'm understanding the flow and that not making the request was intentional, given this is happening in the ApplicationRoute.

Reply:
Yup, exactly, I don't want to fetch any new repos, just get anything that's loaded already and react to any changes.

Reply:
👍